src/integTest/kotlin/io/snyk/plugin/TestUtils.kt

```diff
@@ -8,3 +15,51 @@ fun setupDummyCliFile() {
         cliFile.createNewFile()
     }
 }
+
+fun removeDummyCliFile() {
+    val cliFile = getCliFile()
+    if (cliFile.exists()) {
+        cliFile.delete()
+    }
+}
+
+fun resetSettings(project: Project?) {
```

Comment:
As discussed, we should have a look at just unregistering and re-registering each stateful service before each test. You could cast `getPicoContainer()` to `MutablePicoContainer` and then register/unregister. see https://www.programcreek.com/java-api-examples/?api=org.picocontainer.MutablePicoContainer

Reply:
Actually, Jetbrains provides a Utility Class for that :). https://upsource.jetbrains.com/idea-ce/file/idea-ce-8f0275fd7faaeafdb8900147eab3d256fe4221cb/platform/testFramework/src/com/intellij/testFramework/ServiceContainerUtil.kt?_ga=2.186386601.1807346972.1636115617-321360060.1633965402

Reply:
I like the idea. And thanks for finding ServiceContainerUtil (why do you delete that comment?). It looks like exactly what we need for such service's replacement.
I just feel such deep Integ tests refactoring is way out of scope for that ticket.
Also feel like we need to do some research how (and if) such replacement works. In the past we had problem with SnykApplicationSettingsStateService replacement. Probably due to it PersistentStateComponent nature, but not sure...
So would propose to explore that test's reset way in next tickets/PRs

Reply:
Dunno why it got away...

Reply:
Hmm... surprisingly it works. Hope we don't create another "time bomb" by doing such replacement...
PS @pavel-snyk  we should consider to clean up the same way in the future at least SnykToolWindowPanel as it holds state of cached scan results.